core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorRecordSetProvider.java

```diff
@@ -13,10 +13,36 @@
  */
 package io.trino.spi.connector;
 
+import io.trino.spi.predicate.TupleDomain;
+
 import java.util.List;
 
 public interface ConnectorRecordSetProvider
 {
+    default RecordSet getRecordSet(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorSplit split,
+            ConnectorTableHandle table,
+            List<? extends ColumnHandle> columns,
+            DynamicFilter dynamicFilter)
+    {
+        // By default, ignore dynamic filter (as it is an optimization and doesn't affect correctness).
+        return getRecordSet(transaction, session, split, table, columns, dynamicFilter.getCurrentPredicate());
+    }
+
+    default RecordSet getRecordSet(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorSplit split,
+            ConnectorTableHandle table,
+            List<? extends ColumnHandle> columns,
+            TupleDomain<ColumnHandle> dynamicFilter)
```

Comment:
This `TupleDomain` variant in `ConnectorPageSourceProvider` is deprecated (it pre-dates `DynamicFilter`). We should only have a single method `DynamicFilter`, which contains `DynamicFilter`.

Reply:
Hi @electrum thanks for the review. The reason I added it here is https://github.com/trinodb/trino/blob/fe608f2723842037ff620d612a706900e79c52c8/core/trino-main/src/main/java/io/trino/split/RecordPageSourceProvider.java#L48 
Maybe I should change all places?

Reply:
@jerryleooo i got rid of the deprecated APIs in ConnectorPageSourceProvider in https://github.com/trinodb/trino/pull/8077 to avoid confusion around this

Reply:
Thanks, @raunaqmorarka , then I may just wait for a while. If #8077 needs more time to merge, I will just leave a TODO here.